src/Widgets/Window.vala

```diff
@@ -270,8 +270,53 @@ public class Window : Gtk.Window {
         return false;
     }
 
+    private string[] process_frontmatter (string raw_mk, out string processed_mk) {
+        string[] map = {};
+
+        processed_mk = null;
+
+        // Parse frontmatter
+        if (raw_mk.length > 4 && raw_mk[0:4] == "---\n") {
+            int i = 0;
+            bool frontmatter_closed = true;
+            int last_newline = 3;
+            int next_newline;
+            string line = "";
+            do {
+                next_newline = raw_mk.index_of_char('\n', last_newline + 1);
+                if (next_newline == -1) { // End of file
+                    frontmatter_closed = false;
+                    break;
+                }
+                line = raw_mk[last_newline+1:next_newline];
+
+                var sep_index = line.index_of_char(':');
+                if (sep_index != -1) {
+                    map += line[0:sep_index-1];
+                    map += line[sep_index+1:line.length];
+                }
```

Comment:
there should be an else here so if the header doesn't have a colon we don't process it.

The 'Front Matter' feature needs a colon to separate variable name from value and since there is no colon, the doc isn't using the 'Front Matter' feature.